fast-af/fast-checkout-woocommerce
# Add option to hide button by product

# Description

This adds an option in the Fast settings page to select specific products for which the Fast Checkout button should be hidden.

# Testing instructions

1. Login at https://fasttestdev.wpcomstaging.com/wp-admin/
2. Go to https://fasttestdev.wpcomstaging.com/wp-admin/admin.php?page=fast
3. Scroll down to the setting labeled "Hide Buttons for these Products".
4. Start typing the name of a product, and select the product from the list that pops up.
5. Save the settings.
6. Visit that product's details page on the front end of the site.
7. Verify that the Fast Checkout button is not displayed for that product.
8. Add the product to the cart.
9. Verify that the Fast Checkout button is not displayed in the mini cart after the product is added.
10. Verify that the Fast Checkout button is not displayed in the full cart after the product is added.
11. Verify that the Fast Checkout button is not displayed in the checkout after the product is added to cart.
12. Go to https://fasttestdev.wpcomstaging.com/wp-admin/admin.php?page=fast
13. Remove the product from the "Hide Buttons for these Products" setting.
14. Save the settings.
15. Verify that the Fast Checkout button is displaying as expected.

Note that the mini cart is cached, so it might not show or hide the Fast Checkout button as expected until you add or remove a product from the cart to clear your local cache.

# Checks

@brikr @ilkerulutas this is ready for code review.
